### PR TITLE
manifest: nanopb: Upmerge to version 0.4.6

### PR DESCRIPTION
**Pull request by pdgendt**
This PR upmerges the nanopb module to the tagged 0.4.6 version.

~~A maximum v3 python protobuf package version is set to fix builds with upgraded `scripts/requirements.txt`~~

### CODE REVIEW

File: scripts/requirements-extras.txt
Comment (yperess) [inline scripts/requirements-extras.txt:24]: Am I understanding correctly that this prevents the use of v4.* of protobuf since it has a breaking change? Will this go away once the pull request is merged?

File: scripts/requirements-extras.txt
Comment (mbolivar-nordic) [inline scripts/requirements-extras.txt:22]: The commit message says v4.20.0, but this says v4.21.0. What's going on?

File: scripts/requirements-extras.txt
Comment (mbolivar-nordic) [inline scripts/requirements-extras.txt:24]: Why not just say `<v4.2x.0` for whatever value of `x` is correct?

**Review by mbolivar-nordic — DISMISSED** [review]

File: scripts/requirements-extras.txt
Comment (pdgendt) [reply scripts/requirements-extras.txt:22]: My bad.

File: scripts/requirements-extras.txt
Comment (pdgendt) [reply scripts/requirements-extras.txt:24]: I thought `<4` was cleaner, as only the major part is relevant here.

**Comment by carlescufi** [comment]
@pdgendt  SHA: dc4deed54fd4c7e1935e3b6387eedf21bb45dc38

File: scripts/requirements-extras.txt
Comment (yperess) [reply scripts/requirements-extras.txt]: OK, so I did understand correctly that this is a temporary fix? If so I'm happy with it, can we just make sure there's an issue tracking this so we can fix it up later and make sure that the release team is aware of it (@fabiobaltieri)

Comment (pdgendt) [reply scripts/requirements-extras.txt]: There was a discussion when the `nanopb` was introduced in zephyr's PR #33211. If `grpcio-tools` works on windows, this might be the current preferred way.
This could also eliminate the requirement to have `protoc` installed in the build system. @carlescufi 

Otherwise we would have to work with this temporary requirement. I opened a [nanopb issue](https://github.com/nanopb/nanopb/issues/787) where the maintainer also suggests to use `grpcio-tools`.

**Review by yperess — APPROVED** [review]

**Comment by pdgendt** [comment]
@mbolivar-nordic any open comments?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
